app/src/main/java/de/danoeh/antennapod/fragment/AddFeedFragment.java

```diff
@@ -140,7 +141,13 @@ private void showAddViaUrlDialog() {
         alertViewBinding.urlEditText.setHint(R.string.add_podcast_by_url_hint);
 
         ClipboardManager clipboard = (ClipboardManager) getContext().getSystemService(Context.CLIPBOARD_SERVICE);
-        String clipboardContent = clipboard.getText() != null ? clipboard.getText().toString() : "";
+        String clipboardContent;
+        final ClipData clipData = clipboard.getPrimaryClip();
+        if (clipData != null && clipData.getItemCount() > 0 && clipData.getItemAt(0).getText() != null) {
+            clipboardContent = clipData.getItemAt(0).getText().toString();
+        } else {
+            clipboardContent = "";
+        }
         if (clipboardContent.trim().startsWith("http")) {
             alertViewBinding.urlEditText.setText(clipboardContent.trim());
         }
```

Comment:
If there now is an `if` statement anyway, we can also just put the check in there and reduce the number of lines a bit :)

```suggestion
        final ClipData clipData = clipboard.getPrimaryClip();
        if (clipData != null && clipData.getItemCount() > 0 && clipData.getItemAt(0).getText() != null) {
            String clipboardContent = clipData.getItemAt(0).getText().toString();
            if (clipboardContent.trim().startsWith("http")) {
                alertViewBinding.urlEditText.setText(clipboardContent.trim());
            }
        }
```

Reply:
Didn't occur to me that I didn't need to do `= ""` lol. This looks much better 👍